thrust/thrust/detail/random_bijection.h

```diff
@@ -0,0 +1,173 @@
+/*
+ *  Copyright 2008-2025 NVIDIA Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+//! \file random_bijection.h
+//! \brief An implementation of a bijective function for use in shuffling
+
+#pragma once
+
+#include <thrust/detail/config.h>
+
+#include <thrust/random.h>
+
+#include <cuda/std/cstdint>
+#include <cuda/std/type_traits>
+
+THRUST_NAMESPACE_BEGIN
+namespace detail
```

Comment:
Should this be released as a public interface. My initial thought is no.
In favour is it would allow people to BYO bijection for use with the `random_bijection` class (e.g. an LCG bijection -> a bit faster, worse quality).
Against is that `random_bijection` is sensitive to the quality of the bijection. There are sizes for LCG that will consistently hit one of the load imbalance scenarios. The interface is also not particularly general

Reply:
I think leaving it as private is fine for now. We can easily expose it later if desirable.